File: updated_code/code/python/1652-defuse-the-bomb.py
from typing import List

class Solution:
    """
    Solves the problem of defusing the bomb by decrypting the code.
    
    Parameters:
    code (List[int]): The circular array of length n.
    k (int): The key to decrypt the code.
    
    Returns:
    List[int]: The decrypted code to defuse the bomb.
    """
    def decrypt(self, code: List[int], k: int) -> List[int]:
        """
        Decrypts the given code by replacing each number with the sum of the next k numbers.
        
        Parameters:
        code (List[int]): The circular array of length n.
        k (int): The key to decrypt the code.
        
        Returns:
        List[int]: The decrypted code to defuse the bomb.
        """
        n = len(code)  # Get the length of the code array
        result = []  # Initialize the result list
        
        # Loop through each element in the code array
        for i in range(n):
            # Calculate the sum of the next k numbers
            if k > 0:
                total = sum(code[(i + j) % n] for j in range(1, k + 1))
            # Calculate the sum of the previous k numbers
            elif k < 0:
                total = sum(code[(i - j) % n] for j in range(1, -k) if j != i)
            # Replace each number with 0 when k is 0
            else:
                total = 0
            
            # Add the total to the result list
            result.append(total)
        
        # Return the decrypted code
        return result

# Example usage
code = [5, 7, 1, 4]
k = 3
print(Solution().decrypt(code, k))  # Output: [12, 10, 16, 13]

File: updated_code/code/java/1652-defuse-the-bomb.java
package com.code.java;

import java.util.Arrays;

public class Solution {

    public int[] decrypt(int[] code, int k) {
        if (code == null || code.length == 0) {
            return new int[0];
        }

        int n = code.length;
        int[] result = new int[n];
        int sum = 0;

        for (int i = 0; i < n; i++) {
            sum = 0;
            for (int j = 1; j <= Math.abs(k); j++) {
                int index = (i + j - 1) % n;
                sum += code[index];
            }
            result[i] = k > 0 ? sum : -sum;
        }

        return result;
    }

    public static void main(String[] args) {
        Solution solution = new Solution();
        int[] code = {5, 7, 1, 4};
        int k = 3;
        int[] result = solution.decrypt(code, k);
        System.out.println(Arrays.toString(result));
    }
}